processor/src/main/java/com/log/processor/LogProcessorModule.java

```diff
@@ -73,19 +78,25 @@ private ExecutorService buildExecutor() {
 	
 	@Provides
 	@Inject
-	private List<LogFileProcessor> buildLogFileProcessors(
+	private List<AbstractLogProcessor> buildLogFileProcessors(
 			@Named("numJobs") int numJobs, Injector injector ) {
-		List<LogFileProcessor> readers = new ArrayList<>();
-		for (int job=0; job < numJobs; job++) {
-			readers.add( injector.getInstance(LogFileProcessor.class) );
+
```

Comment:
create separate methods one for file processing and one for s3 processing. Call the one needed based on config value. Create a separate boolean config param for this. For example: say, s3Access. When it is false we will do file processing. When it is true we will do s3 processing.